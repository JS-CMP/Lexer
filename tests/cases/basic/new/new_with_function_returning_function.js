function Factory() {
    return function Product() {
        this.value = 123;
    };
}

new (Factory())(); // NEW(Factory())
