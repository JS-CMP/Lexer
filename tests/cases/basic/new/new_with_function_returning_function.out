#include "types/JsAny.hpp"
#include "global/global.hpp"
#include "types/objects/Types.hpp"
#include "customOperators/CustomOperators.hpp"

int main() {JS::Any Factory;Factory = JS::Any(std::make_shared<JS::Function>([&](const JS::Any &thisArg, const JS::Any &arguments) -> JS::Any {return JS::Any(std::make_shared<JS::Function>([&](const JS::Any &thisArg, const JS::Any &arguments) -> JS::Any {thisArg[u"value"]=JS::Any(123);return JS::Any();}));return JS::Any();}));console[u"log"](NEW((Factory())));}
